physlr/physlr.py

```diff
@@ -1294,12 +1294,28 @@ def physlr_report_junctions_graph(self):
         g = self.read_graph(self.args.FILES)
         gmst = Physlr.determine_pruned_mst(g)
         print(int(timeit.default_timer() - t0), "Searching for junctions...", file=sys.stderr)
-        junctions = []
+        tree_junctions = []
         for component in nx.connected_components(gmst):
-            junctions += Physlr.detect_junctions_of_tree(
+            tree_junctions += Physlr.detect_junctions_of_tree(
                 gmst.subgraph(component), Physlr.args.prune_junctions)
         print(int(timeit.default_timer() - t0),
-              "Found", len(junctions), "junctions.", file=sys.stderr)
+              "Found", len(tree_junctions), "junctions.", file=sys.stderr)
+        junctions = []
+        if self.args.junction_depth > 0:
+            print(int(timeit.default_timer() - t0),
+                  "Exapnding junctions, depth:", self.args.junction_depth, ".", file=sys.stderr)
+            tree_junctions_expanded = []
+            for tree_junction in tree_junctions:
+                tree_junctions_expanded += list(nx.bfs_tree(
+                    gmst, source=tree_junction, depth_limit=self.args.junction_depth))
+            tree_junctions_expanded = list(set(tree_junctions_expanded))
```

Comment:
Avoid converting from a list to a set back to a list. `tree_junctions_expanded` should be a `set` while it's being constructed, then convert to a `list` at the end if you need a `list`, or maybe you don't need a `list` and it can be left as a `set`.